### PR TITLE
feat(vscode.dev): add presence

### PR DESCRIPTION
**Pull request by Faelayis**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image
image

</details>

### CODE REVIEW

File: websites/V/vscode.dev/presence.ts [hunk 1]
```diff
@@ -0,0 +1,626 @@
...
+ * Custom output
+ * - '%file%' (presence.ts)
+ * - '%branch%' (main)
+ * - '%error%' (0)
+ * - '%problems%' (0)
+ * - '%workspace%' (Presences)
+ * - '%lang%' (typescript)
+ * - '%encoding%' (UTF-8)
+ * - '%selection%' (Ln 20, Col 20)
+ */
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/presence.ts:29]: You can create a README.md file like Custom Status presence and add these customisations there

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/V/vscode.dev/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+This is a string that can be used (there must be a `%` sign before and after every time) <br>
+
+- `%file%` will be replaced with the current file name.
+- `%branch%` will be replaced with the current branch name. (remote URL only)
+- `%error%` will be replaced with error current workspace folder.
+- `%problems%` will be replaced with problems current workspace folder.
+- `%workspace%` will be replaced with the current workspace name.
+- `%lang%` will be replaced with the language current file.
+- `%encoding%` will be replaced with the encoding.
+- `%selection%` will be replaced with the current line.
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/README.md:12]: I would use a [markdown table](https://www.tablesgenerator.com/markdown_tables) for this with the headers "Parameter", "Description" and "Example"

File: websites/V/vscode.dev/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# How to custom string
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/README.md:1]:
```suggestion
# Presence Parameters for Customisation
```

File: websites/V/vscode.dev/presence.ts [hunk 4]
```diff
@@ -0,0 +1,626 @@
...
+			presence.getSetting<string>("customImageText"),
+			presence.getSetting<string>("customEmpty"),
+			presence.getSetting<boolean>("timeStamps"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		presenceData: PresenceData = {
+			largeImageKey:
+				"https://raw.githubusercontent.com/leonardssh/vscord/main/assets/icons/idle-vscode.png",
+			smallImageKey:
+				"https://raw.githubusercontent.com/leonardssh/vscord/main/assets/icons/vscode.png",
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/presence.ts:553]: Ditto

File: websites/V/vscode.dev/presence.ts [hunk 5]
```diff
@@ -0,0 +1,626 @@
...
+				?.getAttribute("title"),
+			editor: {
+				lang: document.querySelector("#status\\.editor\\.mode")?.textContent,
+				encoding: document.querySelector("#status\\.editor\\.encoding")
+					?.textContent,
+				selection: document.querySelector("#status\\.editor\\.selection")
+					?.textContent,
+			},
+		},
+		filename = status.file,
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/presence.ts:576]: Update referencds too
```suggestion
		{ file } = status,
```

File: websites/V/vscode.dev/presence.ts [hunk 6]
```diff
@@ -0,0 +1,626 @@
...
+		findExtension = Object.keys(KNOWN_EXTENSIONS).find(key => {
+			if (filename?.endsWith(key)) return true;
+			const match = /^\/(.*)\/([mgiy]+)$/.exec(key);
+			if (!match) return false;
+			return new RegExp(match[1], match[2]).test(filename);
+		}),
+		Replace = function (value: string) {
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/presence.ts:583]: Create a proper [function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions)

File: websites/V/vscode.dev/presence.ts [hunk 7]
```diff
@@ -0,0 +1,626 @@
...
+				.replace("%workspace%", status.workspace || customEmpty)
+				.replace("%lang%", status.editor.lang || customEmpty)
+				.replace("%encoding%", status.editor.encoding || customEmpty)
+				.replace("%selection%", status.editor.selection || customEmpty);
+		};
+
+	if (!status.file || !status.workspace) {
+		presenceData.details = detailIdling;
+		presenceData.largeImageKey =
+			"https://raw.githubusercontent.com/leonardssh/vscord/main/assets/icons/idle-vscode.png";
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/presence.ts:598]: Ditto

File: websites/V/vscode.dev/presence.ts [hunk 8]
```diff
@@ -0,0 +1,626 @@
...
+		if (presenceData.buttons) delete presenceData.buttons;
+	} else {
+		presenceData.state = Replace(customState);
+		presenceData.details = Replace(customDetail);
+		presenceData.smallImageText = Replace(customSmallText);
+		presenceData.largeImageKey = `https://raw.githubusercontent.com/leonardssh/vscord/main/assets/icons/${
```
Comment (EncryptedDev) [inline websites/V/vscode.dev/presence.ts:604]: Ditto

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/V/vscode.dev/presence.ts [hunk 9]
```diff
@@ -0,0 +1,626 @@
...
+			presence.getSetting<string>("customDetail"),
+			presence.getSetting<string>("customState"),
+			presence.getSetting<string>("customImageText"),
+			presence.getSetting<string>("customEmpty"),
+			presence.getSetting<boolean>("timeStamps"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		presenceData: PresenceData = {
+			largeImageKey:
+				"https://raw.githubusercontent.com/leonardssh/vscord/main/assets/icons/idle-vscode.png",
```
[the remark this answers is not in the document]
Comment (Faelayis) [reply websites/V/vscode.dev/presence.ts]: I moved assets to discord art assets

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**